x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_keys_grid_page.tsx

```diff
@@ -429,97 +512,95 @@ export const ApiKeysTable: FunctionComponent<ApiKeysTableProps> = ({
-      ),
-    });
-  }
-
-  if (usernameFilters.length > 1) {
+  if (typeFilters.size > 1) {
     filters.push({
       type: 'custom_component',
       component: ({ query, onChange }) => (
-        <UsersFilterButton usernames={usernameFilters} query={query} onChange={onChange} />
+        <TypesFilterButton types={[...typeFilters]} query={query} onChange={onChange} />
       ),
     });
   }
 
-  if (expiredFilters.length > 1) {
+  if (expired > 0) {
     filters.push({
       type: 'field_value_toggle_group',
-      field: 'expired',
+      field: 'expiration',
       items: [
         {
-          value: false,
+          value: 'now+1m/m',
           name: i18n.translate('xpack.security.management.apiKeys.table.activeFilter', {
             defaultMessage: 'Active',
           }),
+          operator: 'gt',
         },
         {
-          value: true,
+          value: 'now',
           name: i18n.translate('xpack.security.management.apiKeys.table.expiredFilter', {
             defaultMessage: 'Expired',
           }),
+          operator: 'lte',
         },
       ],
     });
   }
 
+  if (usernameFilters.size > 1) {
+    filters.push({
+      type: 'custom_component',
+      component: ({ query, onChange }) => (
+        <UsersFilterButton usernames={[...usernameFilters]} query={query} onChange={onChange} />
+      ),
+    });
+  }
+
   return (
-    <EuiInMemoryTable
-      items={categorizedApiKeys}
-      itemId="id"
-      columns={columns}
-      search={
-        categorizedApiKeys.length > 0
-          ? {
-              toolsLeft: selectedItems.length ? (
-                <EuiButton
-                  onClick={() => onDelete(selectedItems)}
-                  color="danger"
-                  iconType="trash"
-                  data-test-subj="bulkInvalidateActionButton"
-                >
-                  <FormattedMessage
-                    id="xpack.security.management.apiKeys.table.invalidateApiKeyButton"
-                    defaultMessage="Delete {count, plural, one {API key} other {# API keys}}"
-                    values={{
-                      count: selectedItems.length,
-                    }}
-                  />
-                </EuiButton>
-              ) : undefined,
-              box: {
-                incremental: true,
-              },
-              filters,
-            }
-          : undefined
-      }
-      sorting={{
-        sort: {
-          field: 'creation',
-          direction: 'desc',
-        },
-      }}
-      selection={
-        readOnly
-          ? undefined
-          : {
-              selectable: deletable,
-              onSelectionChange: setSelectedItems,
-            }
-      }
-      pagination={{
-        initialPageSize: 10,
-        pageSizeOptions: [10, 25, 50],
-      }}
-      loading={loading}
-      isSelectable={canManageOwnApiKeys}
-    />
+    <>
+      <EuiSearchBar
+        defaultQuery={initialQuery}
+        box={{
+          placeholder: 'Search...',
```

Comment:
There _could_ be an additional prop here called `incremental` that calls `onSearch` while the user is typing. Although this might be helpful, this setting causes the table to flicker between `no results` and `some results` as the query sent to the server _could_ return 0 api keys. 

An alternative could be to debounce onSearch - would like to know your feedback about it.